### PR TITLE
web: add `with_prevent_default`, `with_focusable`

### LINKED ISSUE #1768: [web_sys] Standard web hotkeys are prevented
When I'm on the web, I expect to be able to use Ctrl+R, F5, F12 etc. 
I can understand a developer might want to hijack some keys, but that can easily be done using javascript in the host document. 
The way it is now, it is not possible to get the default behaviour (working F5) using winit. 

#1576 should be reverted.

### PR DESCRIPTION
**Pull request by grovesNL**
`with_prevent_default` controls whether `event.preventDefault` is called

`with_focusable` controls whether `tabindex` is added

Fixes #1768

- [X] Tested on all platforms changed
- [X] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [X] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Review by maroider — COMMENTED** [review]
This seems like a sensible addition on the whole.

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,9 +61,10 @@ And please only add new entries to the top of this list, right below the `# Unre
 - Implemented `Eq` for `Fullscreen`, `Theme`, and `UserAttentionType`.
 - **Breaking:** `Window::set_cursor_grab` now accepts `CursorGrabMode` to control grabbing behavior.
 - On Wayland, add support for `Window::set_cursor_position`.
-- Fix on macOS `WindowBuilder::with_disallow_hidpi`, setting true or false by the user no matter the SO default value. 
+- Fix on macOS `WindowBuilder::with_disallow_hidpi`, setting true or false by the user no matter the SO default value.
```
[the remark this answers is not in the document]
Comment (grovesNL) [reply CHANGELOG.md:64]: Yeah I see linter warnings for trailing white space like this, so I usually format on save. I can remove this change if we'd prefer to leave it as is though.

[the remark this answers is not in the document]
Comment (maroider) [reply CHANGELOG.md:64]: Either way is fine by me for now. I think we should probably get some sort of markdown linting in CI at some point.

[the remark this answers is not in the document]
Comment (grovesNL) [reply CHANGELOG.md:67]: Sounds good, I think we can probably keep the change in that case. I just split it into a separate commit to separate it from the feature.

**Review by maroider — APPROVED** [review]

**Pull request merged**